packages/cheqd/src/dids/didCheqdUtil.ts

```diff
@@ -0,0 +1,152 @@
+import type { DidDocument } from '@aries-framework/core'
+import type { CheqdNetwork, DIDDocument, MethodSpecificIdAlgo, TVerificationKey } from '@cheqd/sdk'
+import type { Metadata } from '@cheqd/ts-proto/cheqd/resource/v2'
+
+import { AriesFrameworkError, JsonEncoder, TypedArrayEncoder } from '@aries-framework/core'
+import {
+  createDidPayload,
+  createDidVerificationMethod,
+  createVerificationKeys,
+  DIDModule,
+  VerificationMethods,
+} from '@cheqd/sdk'
+import { MsgCreateDidDocPayload, MsgDeactivateDidDocPayload } from '@cheqd/ts-proto/cheqd/did/v2'
+import { EnglishMnemonic as _ } from '@cosmjs/crypto'
+import { DirectSecp256k1HdWallet, DirectSecp256k1Wallet } from '@cosmjs/proto-signing'
+
+export function validateSpecCompliantPayload(didDocument: DidDocument): SpecValidationResult {
```

Comment:
Is it supposed to be the same static validation process as on ledger side? If yes. then I think we need to add here a static validation for publicKey for each verification method and not only check for empty.
BTW, do we have the validation on `cheqd/sdk`? I think if it's needed it should on sdk side, not here.

Reply:
No it's not ledger validation that's happening here, it validates a did doc according to the did-core spec. 

Ledger validations are handled in @cheqd/sdk